### ISSUE TITLE
[BUG] in Rancher Logging, server return 500 in response but UI still show success

### ISSUE THREAD

**Issue by aiwantaozi** [rancher/rancher#27695]
**Steps to reproduce (least amount of steps as possible):**
Version: master
Steps:
1. go to cluster logging page -> edit as file
2. input configure, and click dry-run
```
        @type elasticsearch
	include_tag_key  true
	hosts http://18.163.101.228:9200    
	logstash_prefix "ec2-2"
	logstash_format true
	logstash_dateformat  %Y-%m-%d
	type_name  "container_log"
```
3. server crash and return 500, but UI still show success

Related Issue:
https://github.com/rancher/rancher/issues/26543

**Comment by izaac** [comment]
Reproduced in Rancher `v2.4.5`

**Comment by izaac** [comment]
This is what I get in 
- `v2.4-head (06/24/2020)` commit id: 46f7940dd
- `master-head (06/24/2020)` commit id: 0d7c892c0

![Screen Shot 2020-06-24 at 11 13 29 AM (2)](https://user-images.githubusercontent.com/27659/85610874-41eaef80-b60c-11ea-986b-4162e4d64eed.png)

This is much better indicative that something went wrong after clicking `Dry Run`.

The 500 returned by Rancher is getting fixed in a separate issue #26543 so the positive scenario (Dry Run with right settings)   will be scoped there.
